src/main/groovy/com/calgaryscientific/gradle/VeracodeWorkflow.groovy

```diff
@@ -0,0 +1,135 @@
+/*******************************************************************************
+ * MIT License
+ *
+ * Copyright (c) 2017 Calgary Scientific Incorporated
+ *
+ * Copyright (c) 2013-2014 kctang
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ ******************************************************************************/
+
+package com.calgaryscientific.gradle
+
+import groovy.transform.CompileStatic
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+@CompileStatic
+class VeracodeWorkflow {
+    private static Logger log = LoggerFactory.getLogger(VeracodeWorkflow.class);
+
+    static void appWorkflow(VeracodeAPI veracodeAPI,
```

Comment:
I can see we have lots of duplication in appWorkflow here and in sandboxWorkflow
and based on your comment, comments is one of a good place to extract the block of code to function like
```
getBuildStatusOnce
createBuildIfPreviousWorkIsDone
uploadBuildsAndStartPreScan
startScanWithPreScanResults
```

Reply:
I understand, but at the end of the day there is a different method call for sandbox tasks and app tasks so if I do those methods I end up adding an if statement in each method.
I decided to just do each appart.